lib/ansible/modules/cloud/amazon/elb_application_lb.py

```diff
@@ -0,0 +1,849 @@
+      will not be modified.
+    required: false
+    default: yes
+    choices: [ 'yes', 'no' ]
+  subnets:
+    description:
+      - "A list of the IDs of the subnets to attach to the load balancer. You can specify only one subnet per Availability Zone. You must specify subnets from \
+      at least two Availability Zones. Required if state=present."
+    required: false
+  security_groups:
+    description:
+      - A list of the names or IDs of the security groups to assign to the load balancer. Required if state=present.
+    required: false
+    default: []
+  scheme:
+    description:
+      - Internet-facing or internal load balancer. An ELB scheme can not be modified after creation.
+    required: false
+    default: internet-facing
+    choices: [ 'internet-facing', 'internal' ]
+  state:
+    description:
+      - Create or destroy the load balancer.
+    required: true
+    choices: [ 'present', 'absent' ]
+  tags:
+    description:
+      - A dictionary of one or more tags to assign to the load balancer.
+    required: false
+extends_documentation_fragment:
+    - aws
+    - ec2
+'''
+
+EXAMPLES = '''
+# Note: These examples do not set authentication details, see the AWS Guide for details.
+
+# Create an ELB and attach a listener
+- elb_application_lb:
+    name: myelb
+    security_groups:
+      - sg-12345678
+      - my-sec-group
+    subnets:
+      - subnet-012345678
+      - subnet-abcdef000
+    listeners:
+      - Protocol: http # Required. The protocol for connections from clients to the load balancer (http or https).
+        Port: 80 # Required. The port on which the load balancer is listening.
+        SslPolicy: # The security policy that defines which ciphers and protocols are supported. The default is the current predefined security policy.
+        Certificates: # The ARN of the certificate (only one certficate ARN should be provided)
+        DefaultActions:
+          - Type: forward # Required. Only 'forward' is accepted at this time
+            TargetGroupName: # Required. The name of the target group
+    state: present
+
+# Create an ELB and attach a listener with logging enabled
+- elb_application_lb:
+    access_logs_enabled: yes
+    access_logs_s3_bucket: mybucket
+    access_logs_s3_prefix: "/logs"
+    name: myelb
+    security_groups:
+      - sg-12345678
+      - my-sec-group
+    subnets:
+      - subnet-012345678
+      - subnet-abcdef000
+    listeners:
+      - Protocol: http # Required. The protocol for connections from clients to the load balancer (http or https).
+        Port: 80 # Required. The port on which the load balancer is listening.
+        SslPolicy: # The security policy that defines which ciphers and protocols are supported. The default is the current predefined security policy.
+        Certificates: # The ARN of the certificate (only one certficate ARN should be provided)
```

Comment:
The later example of the Certificates parameter is much more clear about what's expected here (a list of dicts, and not a string of the ARN, which is what the comment here made it sound like) -- it might be nice if this example matched.